library/src/scala/quoted/Quotes.scala

```diff
@@ -682,6 +682,22 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       def copy(original: Tree)(name: String, tpt: TypeTree, rhs: Option[Term]): ValDef
       def unapply(vdef: ValDef): (String, TypeTree, Option[Term])
 
+      /** Creates a block `{ val <name> = <rhs: Term>; <body(x): Term> }`
+       *
+       *  Usage:
+       *  ```
+       *  ValDef.let(owner, "x", rhs1, Flags.Lazy) { x =>
+       *    ValDef.let(x.symbol.owner, "y", rhs2, Flags.Mutable) { y =>
+       *      // use `x` and `y`
+       *    }
+       *  }
+       *  ```
+       *
+       *  @param flags extra flags to with which the symbol should be constructed. Can be `Private | Protected | Override | Deferred | Final | Param | Implicit | Lazy | Mutable | Local | ParamAccessor | Module | Package | Case | CaseAccessor | Given | Enum | JavaStatic | Synthetic | Artifact`
```

Comment:
Since we know that this new val is going to be defined in a Block (as opposed to class, object, etc.), we know that less flags/modifiers are going to be allowed there, so ideally we would limit this list here (e.g. access modifiers like `Private`, or `Protected` do not make sense to have here).

Reply:
Is there a defined list where the allowed subset is defined? The only ones that I can think of, or at least that I was trying to use, were `Lazy` and `Mutable`.